app/gui/view/welcome-screen/src/side_menu.rs

```diff
@@ -37,8 +42,31 @@ impl SideMenu {
         let projects_list = Self::create_projects_list();
         root_dom.append_or_warn(&projects_list, &logger);
         let new_project_button = Self::create_new_project_button(&logger, &projects_list);
+        let closures = default();
 
-        Self { logger, root_dom, projects_list, new_project_button, closures: default() }
+        let menu = Self { logger, root_dom, frp, projects_list, new_project_button, closures };
+        menu.setup_new_project_event_listener();
+        menu
+    }
+
+    pub fn set_projects_list(&self, projects: &[String]) {
+        self.clear_projects_list();
+        let new_button = &self.new_project_button;
+        for project in projects {
+            let node = Self::create_project_list_entry(project);
+            self.setup_open_project_event_listener(&node, project);
+            self.projects_list.insert_before_or_warn(&node, new_button, &self.logger);
+        }
+    }
+
+    fn clear_projects_list(&self) {
+        let entries = self.projects_list.children();
+        for i in 0..entries.length() {
+            let entry = entries.item(i).expect("Out of bounds error.");
```

Comment:
Do not create code that can crash the IDE. If something bad happens, this should be logged to the console, but the app should never crash if there is ANY way to keep it working. Also, this error is super cryptic - app crashes and you see "Out of bounds error" - no one knows what happened based on such an error.

Reply:
Yeah, this code wasn't even working correctly. I fixed it.